File: Kattis/solutions/warehouse/warehouse.cpp
#include <iostream>
#include <map>
#include <set>
using namespace std;

int main() {
  int t;
  cin >> t;

  while (t) {
    int n;
    cin >> n;

    map<string, int> frequency;
    for (int i = 0; i < n; i++) {
      string toy;
      int num;
      cin >> toy >> num;

      frequency[toy] += num;
    }

    map<int, set<string>> res;
    for (auto& pair : frequency) {
      res[pair.second].insert(pair.first);
    }

    cout << frequency.size() << "\n";
    for (auto it = res.rbegin(); it != res.rend(); it++) {
      auto& keyValue = *it;
      for (auto& s : keyValue.second) {
        cout << s << " " << keyValue.first << "\n";
      }
    }

    t--;
  }

  return 0;
}

File: Kattis/solutions/warehouse/warehouse.java
import java.util.*;

public class warehouse {
  public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);
    int t = scanner.nextInt();

    while (t > 0) {
      int n = scanner.nextInt();

      Map<String, Integer> frequency = new HashMap<>();
      for (int i = 0; i < n; i++) {
        String toy = scanner.next();
        int num = scanner.nextInt();

        frequency.put(toy, frequency.getOrDefault(toy, 0) + num);
      }

      Map<Integer, Set<String>> res = new TreeMap<>(Collections.reverseOrder());
      for (Map.Entry<String, Integer> entry : frequency.entrySet()) {
        int count = entry.getValue();
        String toy = entry.getKey();

        if (!res.containsKey(count)) {
          res.put(count, new TreeSet<>());
        }

        res.get(count).add(toy);
      }

      System.out.println(frequency.size());
      for (Map.Entry<Integer, Set<String>> entry : res.entrySet()) {
        for (String toy : entry.getValue()) {
          System.out.println(toy + " " + entry.getKey());
        }
      }

      t--;
    }

    scanner.close();
  }
}